File: app/src/main/java/id/ac/uasbungaa/helper/LocalDataHelper.java
package id.ac.uasbungaa.helper;

import android.database.Cursor;

import id.ac.uasbungaa.data.local.BarangHelper;
import id.ac.uasbungaa.data.local.UserHelper;
import id.ac.uasbungaa.data.model.Barang;

import java.util.ArrayList;

import org.jetbrains.annotations.NotNull;

public final class LocalDataHelper {
    @NotNull
    public static final LocalDataHelper INSTANCE;

    @NotNull
    public final ArrayList getAllUser(@NotNull UserHelper userHelper) {
        userHelper.open();
        Cursor users = userHelper.getAllUser();
        return MappingHelper.INSTANCE.mapCursorAllUser(users);
    }

    @NotNull
    public final ArrayList getAllBarang(@NotNull BarangHelper barangHelper) {
        barangHelper.open();
        Cursor barangs = barangHelper.getAllBarang();
        return MappingHelper.INSTANCE.mapCursorAllBarang(barangs);
    }

    @NotNull
    public final ArrayList getAllBarangKode(@NotNull BarangHelper barangHelper) {
        barangHelper.open();
        Cursor kodes = barangHelper.getAllBarangKode();
        return MappingHelper.INSTANCE.mapCursorAllBarangKode(kodes);
    }

    @NotNull
    public final Barang getBarang(@NotNull BarangHelper barangHelper, @NotNull String kodeBarang) {
        barangHelper.open();
        Cursor barang = barangHelper.getBarangByKode(kodeBarang);
        return MappingHelper.INSTANCE.mapCursorBarang(barang);
    }

    private LocalDataHelper() {
    }

    static {
        INSTANCE = new LocalDataHelper();
    }
}

File: app/src/main/java/id/ac/uasbunga/helper/LocalDataHelper.kt
package id.ac.uasbunga.helper

import id.ac.uasbunga.data.local.BarangHelper
import id.ac.uasbunga.data.local.UserHelper
import id.ac.uasbunga.data.model.Barang
import id.ac.uasbunga.data.model.User
import java.lang.Exception

object LocalDataHelper {
    fun getAllUser(userHelper: UserHelper): ArrayList<User> {
        userHelper.open()
        val users = userHelper.getAllUser()
        return MappingHelper.mapCursorAllUser(users)
    }

    fun getAllBarang(barangHelper: BarangHelper): ArrayList<Barang> {
        barangHelper.open()
        val barangs = barangHelper.getAllBarang()
        return MappingHelper.mapCursorAllBarang(barangs)
    }

    fun getAllBarangKode(barangHelper: BarangHelper): ArrayList<String> {
        barangHelper.open()
        val kodes = barangHelper.getAllBarangKode()
        return MappingHelper.mapCursorAllBarangKode(kodes)
    }

    fun getBarang(barangHelper: BarangHelper, kodeBarang: String): Barang {
        barangHelper.open()
        val barang = barangHelper.getBarangByKode(kodeBarang)
        return MappingHelper.mapCursorBarang(barang)
    }
}